opentelemetry-api/tests/context/test_threadlocal_context.py

```diff
@@ -59,3 +62,19 @@ def test_set_value(self):
         self.assertEqual("zzz", context.get_value("a", context=second))
         self.assertEqual("---", context.get_value("a", context=third))
         self.assertEqual(None, context.get_value("a"))
+
+    @patch(
+        "opentelemetry.context._RUNTIME_CONTEXT",  # type: ignore
+        ThreadLocalRuntimeContext(),
+    )
+    def test_set_current(self):
+        context.attach(context.set_value("a", "yyy"))
+
+        token = context.attach(context.set_value("a", "zzz"))
+        self.assertEqual("zzz", context.get_value("a"))
+
+        context.detach(token)
```

Comment:
It might be worth checking that we can detach contexts out of order too, maybe in a separate `test_detach` test:

```python
In [2]: from opentelemetry import context

In [3]: t1 = context.attach(context.set_value('c', 1)); context.get_current()
Out[3]: {'c': 1}

In [4]: t2 = context.attach(context.set_value('c', 2)); context.get_current()
Out[4]: {'c': 2}

In [5]: context.detach(t1); context.get_current()
Out[5]: {}

In [6]: context.detach(t2); context.get_current()
Out[6]: {'c': 1}
```

I was surprised to see contextvars works this way. One difference between contextvars and threadlocals is that contextvars will raise if you try to reset using the same token twice. I don't know whether it's worth enforcing this behavoir for threadlocals though.

Reply:
test added. Didn't know about the reset behaviour either, good thinng to know!